metsim/disaggregate.py

```diff
@@ -75,17 +75,21 @@ def temp(df_daily, df_disagg, t_t_min, t_t_max, ts):
     temp = np.append(np.insert(temp, 0, temp[0:2]), temp[-2:])
 
     # Interpolate the values
-    interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
-    temps = interp(ts * np.arange(0, len(df_disagg.index)))
+    try:
+        interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
+        temps = interp(ts * np.arange(0, len(df_disagg.index)))
+    except ValueError:
+        temps = np.full(len(df_disagg.index), np.nan)
+
     return temps
 
 
-def precip(precip, ts):
+def prec(prec, ts):
     """
     Splits the daily precipitation evenly throughout the day
     """
     scale = int(ts) / (cnst.MIN_PER_HOUR * cnst.HOURS_PER_DAY)
-    return (precip*scale).resample('{:0.0f}T'.format(ts)).fillna(method='ffill')
+    return (prec * scale).resample('{:0.0f}T'.format(ts)).fillna(method='ffill')
```

Comment:
Ideally you would split your PRs into more specific elements - these changes are not associate with "adding parallelism", so they're like unrelated things being tacked onto a bill in congress. The reason it matters is that it is easier to roll back specific PRs (in case something is wrong) if they are more atomic. Leave as-is for now